sql-cli/noxfile.py

```diff
@@ -16,61 +16,35 @@ def dev(session: nox.Session) -> None:
     development environment to ``.nox/dev``.
     """
     session.install("nox")
-    session.install("-e", "../python-sdk/.[all]")
-    session.install("-e", "../python-sdk/.[tests]")
-    session.install("-e", ".[all]")
-    session.install("-e", ".[tests]")
+    session.install("poetry")
+    session.run("poetry", "install")
 
 
 @nox.session(python=["3.7", "3.8", "3.9"])
-@nox.parametrize("airflow", ["2.2.5", "2.3"])
+@nox.parametrize("airflow", ["2.3"])
 def test(session: nox.Session, airflow) -> None:
     """Run both unit and integration tests."""
-    session.install("-e", "../python-sdk/.[all]")
-    session.install("-e", "../python-sdk/.[tests]")
-    session.install("-e", ".[all]")
-    session.install("-e", ".[tests]")
+    session.install("poetry")
+    session.run("poetry", "install")
     # Log all the installed dependencies
     session.log("Installed Dependencies:")
     session.run("pip3", "freeze")
-    session.run("airflow", "db", "init")
-    session.run("pytest", *session.posargs)
+    session.run(
+        "pytest", *session.posargs, "--cov=sql_cli", "--cov-report=xml", "--cov-branch"
+    )
 
 
 @nox.session(python=["3.8"])
 def type_check(session: nox.Session) -> None:
     """Run MyPy checks."""
-    session.install("-e", "../python-sdk/.[all]")
-    session.install("-e", "../python-sdk/.[tests]")
-    session.install("-e", ".[all]")
-    session.install("-e", ".[tests]")
+    session.install("poetry")
+    session.run("poetry", "install")
+    session.install("-e", "../sql-cli/.[tool.poetry.dependencies]")
+    session.install("-e", "../sql-cli/.[tool.poetry.group.dev.dependencies]")
     session.run("mypy", "--version")
     session.run("mypy")
 
 
-@nox.session()
-@nox.parametrize(
-    "extras",
-    [
-        ("postgres-amazon", {"include": ["postgres", "amazon"]}),
-        ("snowflake-amazon", {"include": ["snowflake", "amazon"]}),
-        ("sqlite", {"include": ["sqlite"]}),
-    ],
-)
-def test_examples_by_dependency(session: nox.Session, extras):
```

Comment:
We may want to have something similar for the DAGs generated by the SQL CLI - but that's not part of the scope or the code ATM.